api/bin/chainflip-lp-api/src/main.rs

```diff
@@ -486,27 +486,34 @@ impl RpcServer for RpcServerImpl {
 			.await?)
 	}
 
-	fn subscribe_order_fills(&self, pending_sink: PendingSubscriptionSink) {
-		let state_chain_client = self.api.state_chain_client.clone();
-		tokio::spawn(async move {
-			if let Ok(sink) = pending_sink.accept().await {
-				let mut finalized_block_stream = state_chain_client.finalized_block_stream().await;
-				while let Some(block) = finalized_block_stream.next().await {
-					match order_fills(state_chain_client.clone(), block).await {
-						Ok(order_fills) => {
-							if sink
-								.send(sc_rpc::utils::to_sub_message(&sink, &order_fills))
-								.await
-								.is_err()
-							{
-								break;
-							}
-						},
-						Err(_) => break,
-					}
-				}
-			}
-		});
+	async fn subscribe_order_fills(
+		&self,
+		pending_sink: PendingSubscriptionSink,
+	) -> SubscriptionResult {
+		let sub = self
```

Comment:
👀  this change was reverted in a separate PR, and now it's back again?

Reply:
ah, you need to rebase - and then we want the same for the new broker-api subscription

Reply:
Rebased. And re-implemented the `tainted_events_subscription` to work similar to the lp api.